### PR TITLE
feat: add quit command

### PR DESCRIPTION
**Pull request by theovgl**

### CODE REVIEW

File: include/commands.hpp [hunk 1]
```diff
@@ -2,5 +2,11 @@
 # define COMMANDS_HPP
 
 # include "IRCMessage.hpp"
+# include "IRCPeer.hpp"
+
+struct QuitMessage : public IRCMessage
```
Comment (vfurmane) [inline include/commands.hpp:7]: Should be called `ErrorMessage` instead of `QuitMessage`

File: src/commands.cpp [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#include "commands.hpp"
+
+QuitMessage::ErrorMessage(IRCPeer &peer, const std::string &quit_message)
```
Comment (vfurmane) [inline src/commands.cpp:3]: Should be called `ErrorMessage` instead of `QuitMessage`

File: src/commands/quit.cpp [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#include "commands.hpp"
+
+void	command_quit(IRCMessage &message, Dependencies &deps)
+{
+	deps.peers.closeConnection(message.peer.getFd());
+	message.peer.sendMessage(ErrorMessage(message.peer, message.getArguments()[]));
```
Comment (vfurmane) [inline src/commands/quit.cpp:6]: No index inside the `[]` operator

**Review by vfurmane — CHANGES_REQUESTED** [review]

**Review by ppiques — APPROVED** [review]

**Review by vfurmane — APPROVED** [review]
What a surprise

**Pull request merged**